download-data.py

```diff
@@ -424,6 +461,26 @@ def main() -> None:
             file=sys.stderr,
         )
 
+
+    ### Update symlink if requested, without downloading data, if the target directory exists
+    if args.update_symlink:
+
+        if args.test_data:
+            target = "test"
+        else:
+            target = args.release
+
+        if target == "current":
+            # find the most recent local release to use as target
+            dated_releases = {x.name for x in args.data_dir.iterdir() if x.is_dir and re.match("\d{4}-\d{2}-\d{2}$", x.name)}
+            most_recent = sorted(dated_releases)[-1]
```

Comment:
Since we are not using `in` or any other set operation, no need for a set here. (But since we only use it once, we can use a generator)
```suggestion
            dated_releases = (x.name for x in args.data_dir.iterdir() if x.is_dir and re.match("\d{4}-\d{2}-\d{2}$", x.name)
            most_recent = max(dated_releases)
```